server/Arcadia.Assistant.Web/Download/DownloadActor.cs

```diff
@@ -0,0 +1,176 @@
+            IDownloadApplicationSettings downloadApplicationSettings,
+            IHttpClientFactory httpClientFactory,
+            IHostingEnvironment hostingEnvironment)
+        {
+            this.androidDownloadBuildActor = Context.ActorOf(
+                Props.Create(() => new DownloadApplicationActor(
+                    downloadApplicationSettings,
+                    httpClientFactory,
+                    hostingEnvironment,
+                    downloadApplicationSettings.AndroidGetBuildsUrl,
+                    downloadApplicationSettings.AndroidGetBuildDownloadLinkTemplateUrl)),
+                "download-android-build");
+
+            this.iosDownloadBuildActor = Context.ActorOf(
+                Props.Create(() => new DownloadApplicationActor(
+                    downloadApplicationSettings,
+                    httpClientFactory,
+                    hostingEnvironment,
+                    downloadApplicationSettings.IosGetBuildsUrl,
+                    downloadApplicationSettings.IosGetBuildDownloadLinkTemplateUrl)),
+                "download-ios-build");
+
+            Context.System.Scheduler.ScheduleTellRepeatedly(
+                TimeSpan.Zero,
+                TimeSpan.FromMinutes(downloadApplicationSettings.DownloadBuildIntervalMinutes),
+                this.Self,
+                RefreshApplicationBuilds.Instance,
+                this.Self);
+        }
+
+        public IStash Stash { get; set; }
+
+        protected override void OnReceive(object message)
+        {
+            switch (message)
+            {
+                case GetLatestApplicationBuildPath msg:
+                    this.RespondLatestApplicationBuildPath(msg);
+                    break;
+
+                case RefreshApplicationBuilds _:
+                    this.Self.Tell(RefreshApplicationBuildsStart.Instance);
+                    this.BecomeStacked(this.DownloadingApplicationBuilds);
+                    break;
+
+                default:
+                    this.Unhandled(message);
+                    break;
+            }
+        }
+
+        private void DownloadingApplicationBuilds(object message)
+        {
+            switch (message)
+            {
+                case RefreshApplicationBuildsStart _:
+                    var androidDownloadTask = this.androidDownloadBuildActor
+                        .Ask<DownloadApplicationBuild.Response>(DownloadApplicationBuild.Instance);
+                    var iosDownloadTask = this.iosDownloadBuildActor
+                        .Ask<DownloadApplicationBuild.Response>(DownloadApplicationBuild.Instance);
+
+                    Task.WhenAll(androidDownloadTask, iosDownloadTask).PipeTo(
+                        this.Self,
+                        success: result =>
+                        {
+                            var resultExceptions = new List<Exception>();
+
+                            if (result[0] is DownloadApplicationBuild.Error errAndroid)
```

Comment:
result.Of<DownloadApplicationBuild.Error>().Select(x => x.Exception).ToList()

Reply:
Done.